tests/suites/test_suite_cipher.function

```diff
@@ -750,8 +753,23 @@ void dec_empty_buf(  )
     /* decode 0-byte string */
     TEST_ASSERT( 0 == mbedtls_cipher_update( &ctx_dec, encbuf, 0, decbuf, &outlen ) );
     TEST_ASSERT( 0 == outlen );
-    TEST_ASSERT( MBEDTLS_ERR_CIPHER_FULL_BLOCK_EXPECTED == mbedtls_cipher_finish(
-                 &ctx_dec, decbuf + outlen, &outlen ) );
+    if( cipher_info->mode == MBEDTLS_MODE_CBC ||
+        cipher_info->mode == MBEDTLS_MODE_ECB )
+    {
+        /* CBC and ECB ciphers need a full block of input. */
+        TEST_ASSERT( MBEDTLS_ERR_CIPHER_FULL_BLOCK_EXPECTED ==
```

Comment:
I think it would be more readable to first set `expected_ret` based on the mode, and then make a single call to `mbedtls_cipher_finish` and compare the result with `expected_ret`.